diff -up logrotate-3.7.6/logrotate.c.segfault logrotate-3.7.6/logrotate.c
--- logrotate-3.7.6/logrotate.c.segfault	2008-04-23 14:40:51.000000000 +0200
+++ logrotate-3.7.6/logrotate.c	2008-04-23 14:41:14.000000000 +0200
@@ -1380,6 +1380,7 @@ static int readState(char *stateFilename
     line++;
 
     while (fgets(buf, sizeof(buf) - 1, f)) {
+	argv = NULL;
 	line++;
 	i = strlen(buf);
 	if (buf[i - 1] != '\n') {
