diff -up logrotate-3.7.8/examples/logrotate-default.perm logrotate-3.7.8/examples/logrotate-default
--- logrotate-3.7.8/examples/logrotate-default.perm	2010-01-11 13:48:15.000000000 +0100
+++ logrotate-3.7.8/examples/logrotate-default	2010-01-11 13:48:46.000000000 +0100
@@ -6,7 +6,7 @@ weekly
 rotate 4
 
 # create new (empty) log files after rotating old ones
-create
+create 0640 root adm
 
 # use date as a suffix of the rotated file
 dateext
